Framework/PythonInterface/plugins/algorithms/SNSPowderReduction.py

```diff
@@ -1396,6 +1396,35 @@ def _create_absorption_input(self, filename, material=None, geometry=None, envir
         # this effectively deletes the metadata only workspace
         AnalysisDataService.addOrReplace(absName, absorptionWS)
 
+        # Set ChemicalFormula, and either SampleMassDensity or Mass, if SampleMassDensity not set
+        if material is not None:
+            if (not material['ChemicalFormula']) and ("SampleFormula" in absorptionWS.run()):
+                material['ChemicalFormula'] = absorptionWS.run()['SampleFormula'].lastValue().strip()
+            if ("SampleMassDensity" not in material or not material['SampleMassDensity']) and ("SampleDensity" in absorptionWS.run()):
+                if (absorptionWS.run()['SampleDensity'].lastValue() != 1.0) and (absorptionWS.run()['SampleDensity'].lastValue() != 0.0):
+                    material['SampleMassDensity'] = absorptionWS.run()['SampleDensity'].lastValue()
+                else:
+                    material['Mass'] = absorptionWS.run()['SampleMass'].lastValue()
+
+        # Set height for computing density if height not set
+        if geometry is None:
+            geometry = {}
+
+        if geometry is not None:
+            if "Height" not in geometry or not geometry['Height']:
+                # Check units - SetSample expects cm
+                if absorptionWS.run()['BL11A:CS:ITEMS:HeightInContainerUnits'].lastValue() == "mm":
+                    geometry['Height'] = absorptionWS.run()['BL11A:CS:ITEMS:HeightInContainer'].lastValue()*(0.1)
+                elif absorptionWS.run()['BL11A:CS:ITEMS:HeightInContainerUnits'].lastValue() == "cm":
+                    geometry['Height'] = absorptionWS.run()['BL11A:CS:ITEMS:HeightInContainer'].lastValue()
+                else:
+                    pass
```

Comment:
Not understanding the units should be a `ValueError` rather than silently ignoring the height. The current version will make people search for why the the height isn't being set when supplied an "unusual" file.